sympy/solvers/tests/test_solveset.py

```diff
@@ -3530,3 +3530,14 @@ def test_issue_22628():
 
 def test_issue_25781():
     assert solve(sqrt(x/2) - x) == [0, S.Half]
+
+def stationary_points(function, variable, domain):
+    return set()
+
+def test_issue_26077():
+    x = Symbol('x', real=True)
+    function = x*cot(5*x)
+    result = stationary_points(function, x, S.Reals)
+    expected_result = set()
+    print("Actual Result:", result)
+    print(f"Expected Result: {expected_result}")
```

Comment:
The test code is supposed to test the actual code but here you have added a separate function `stationary_points` that does not do anything and written a test for that. The problem is with the sympy `stationary_points` function so the test should be for that and the result returned from that function would not be `set()` for the given inputs:
```python
In [3]:     x = Symbol('x', real=True)
   ...:     function = x*cot(5*x)
   ...:     result = stationary_points(function, x, S.Reals)

In [4]: result
Out[4]: 
⎧  │         ⎛⎧2⋅n⋅π │      ⎫   ⎧2⋅n⋅π   π │      ⎫⎞   ⎛  ⎛       2         ⎞               ⎞⎫
⎨x │ x ∊ ℝ \ ⎜⎨───── │ n ∊ ℤ⎬ ∪ ⎨───── + ─ │ n ∊ ℤ⎬⎟ ∧ ⎝x⋅⎝- 5⋅cot (5⋅x) - 5⎠ + cot(5⋅x) = 0⎠⎬
⎩  │         ⎝⎩  5   │      ⎭   ⎩  5     5 │      ⎭⎠                                         ⎭

In [5]: repr(result)
Out[5]: 'ConditionSet(x, Eq(x*(-5*cot(5*x)**2 - 5) + cot(5*x), 0), Complement(Reals, Union(ImageSet(Lambda(_n, 2*_n*pi/5), Integers), ImageSet(Lambda(_n, 2*_n*pi/5 + pi/5), Integers))))'
```
Also tests should not print anything. They should use assert like `assert result == expected_result`. The test should be written so that it would fail when run with the master branch but succeeds with the other changes in the PR that fix the result of SymPy's `stationary_points` function.

There is plenty of other test code in this file and in other files throughout the sympy codebase that you can look at to ese how tests are written.

Reply:
@oscarbenjamin I am getting this error

<img width="958" alt="Screenshot 2024-01-16 133206" src="https://github.com/sympy/sympy/assets/97588030/1634e2ec-5172-4b67-a157-1af450138d5a">